### PR TITLE
#Module3

### PR DESCRIPTION
**Pull request by TheElidanXD**
Add sequelize in user service

### CODE REVIEW

File: module3/src/app.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// @ts-ignore
+import { Config } from './config.ts';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import UserController from './controllers/user.controller.ts';
+
+const app = express();
+app.listen(Config.port);
+
+const appRouter = Router();
```
Comment (evgenDer) [inline module3/src/app.ts]: express.Router()

File: module3/src/controllers/user.controller.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+// @ts-ignore
+import UserService from '../services/user.service.ts';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { validateSchema } from '../utils/utils.ts';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { userSchema } from '../validators/user.ts';
+
+const userService = UserService();
```
Comment (evgenDer) [inline module3/src/controllers/user.controller.ts]: If you want to write in oop style please use classes and e.g. But it is so strange to call function this way

File: module3/src/controllers/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+const usersRouter = Router();
+usersRouter.use(express.json());
+const userRouter = Router();
```
Comment (evgenDer) [inline module3/src/controllers/user.controller.ts]: duplicate code

File: module3/src/controllers/user.controller.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+userRouter.use(express.json());
+
+export default (app: Router) => {
+    app.use('/users', usersRouter);
```
Comment (evgenDer) [inline module3/src/controllers/user.controller.ts]: Move it to main file. It is help to understand the routes mapping

File: module3/src/controllers/user.controller.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    usersRouter.use('/user', userRouter);
```
Comment (evgenDer) [inline module3/src/controllers/user.controller.ts]: users/user? I think you should split them to separate ones

File: module3/src/models/user.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+export default (sequelize: Sequelize) => sequelize.define('user', user, {
+    createdAt: false,
+    updatedAt: false
+});
+
+export interface User {
+    id: string,
+    login?: string,
```
Comment (evgenDer) [inline module3/src/models/user.ts]: Can you create user without login or password?

File: module3/src/config.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export const Config = {
+    port: 3000,
+    databaseURL: 'postgres://tbrebkof:ybWL2zwMXJJLQqk8GaW-eWucW0EMa-nm@mel.db.elephantsql.com/tbrebkof'
```
Comment (evgenDer) [inline module3/src/config.ts]: move it to env file. It is a private info

File: module3/src/services/user.service.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+import UserModel, { User } from '../models/user.ts';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { Config } from '../config.ts';
+
+const sequelize = new Sequelize(Config.databaseURL);
+sequelize.authenticate().then(() => {
+    console.log('Connected');
+}).catch(err => {
+    console.log('Connection error:', err);
```
Comment (evgenDer) [inline module3/src/services/user.service.ts]: console.error

**Review by evgenDer — CHANGES_REQUESTED** [review]

File: module3/package.json [hunk 9]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "module_3_hw",
+  "version": "1.0.0",
+  "description": "module 3 hw",
+  "main": "index.js",
+  "scripts": {
+    "task1": "ts-node --esm src/app.ts"
```
Comment (evgenDer) [inline module3/package.json]: To fix all your issues with imports use: "node --experimental-specifier-resolution=node --loader ts-node/esm ./src/app.ts"

File: module3/src/services/user.service.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+});
+
+export default () => {
+    const user = UserModel(sequelize);
+    return {
+        getAllUsers: () => user.findAll(
+            { where: { isdeleted: false } }
+        ),
+        createUser: (newUser: User) => user.create(newUser),
```
Comment (evgenDer) [inline module3/src/services/user.service.ts]: type error

File: module3/src/app.ts (hunk 1 above)
Comment (TheElidanXD) [reply module3/src/app.ts]: Done

File: module3/src/controllers/user.controller.ts (hunk 2 above)
Comment (TheElidanXD) [reply module3/src/controllers/user.controller.ts]: Moved to UserService class

File: module3/src/controllers/user.controller.ts (hunk 3 above)
Comment (TheElidanXD) [reply module3/src/controllers/user.controller.ts]: I did not fully understand what in this case a duplicate was.
Can you please clarify?

File: module3/src/controllers/user.controller.ts (hunk 4 above)
Comment (TheElidanXD) [reply module3/src/controllers/user.controller.ts]: Done

File: module3/src/controllers/user.controller.ts (hunk 5 above)
Comment (TheElidanXD) [reply module3/src/controllers/user.controller.ts]: Done

File: module3/src/models/user.ts (hunk 6 above)
Comment (TheElidanXD) [reply module3/src/models/user.ts]: Yes, you are right, these fields are required. Interface fixed.

File: module3/src/controllers/user.controller.ts (hunk 3 above)
Comment (evgenDer) [reply module3/src/controllers/user.controller.ts]: My bad it's ok

File: module3/src/config.ts (hunk 7 above)
Comment (TheElidanXD) [reply module3/src/config.ts]: Moved to .env

File: module3/src/services/user.service.ts (hunk 8 above)
Comment (TheElidanXD) [reply module3/src/services/user.service.ts]: Done

File: module3/src/services/user.service.ts (hunk 10 above)
Comment (TheElidanXD) [reply module3/src/services/user.service.ts]: Fixed

File: module3/package.json (hunk 9 above)
Comment (TheElidanXD) [reply module3/package.json]: It really worked. Thanks a lot 🙏

**Pull request merged**